File: AngularYSpring/ErosketaAngular/src/app/services/cart/cart.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core'
import { Observable, of } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Cart } from '../../classes (interfaces)/cart';
import { CartRequest } from '../../classes (interfaces)/cart-request';


@Injectable({
  providedIn: 'root'
})


export class CartService {

  constructor(private httpCliente: HttpClient) {}

  getListByUser(email: string): Observable<Cart[]> {
    return this.httpCliente.get<Cart[]>(environment.urlCart + "/" + email)
  }

  countByClient(email: string): Observable<number> {
    return this.httpCliente.get<number>(environment.urlCart + "/count/" + email)
  }

  updateProductQuantity(userEmail: string, productId: number, quantity: number): Observable<any> {
    return this.httpCliente.get(environment.urlCart+"/updateQuantity/"+userEmail+"/"+productId+"/"+quantity);
  }

  addProduct(cart: CartRequest): Observable<any> {
    return this.httpCliente.post<any>(environment.urlCart, cart)
  }

  removeProduct(productId: number): Observable<Object> {
    return this.httpCliente.delete(environment.urlCart + "/clean/" + productId)
  }

  removeAllProducts(email: string): Observable<Object> {
    return this.httpCliente.delete(environment.urlCart + "/cleanAll/" + email)
  }

  calculateTotalCart(email: string): Observable<number>{
    return this.httpCliente.get<number>(environment.urlCart + "/totalPrice/" + email)
  }
}

File: AngularYSpring/ErosketaSpringBoot/src/main/java/ies/joatzel/erosketa/Fservices/CartService/CartService.java
package ies.joatzel.erosketa.Fservices.CartService;

import ies.joatzel.erosketa.Amodels.Cart;


public interface CartService {
    java.util.List<Cart> getListByUser(String email);
    void cleanCart(String email);
    void removeProduct(Long productId);
    void addProduct(Cart cart);
    Long getCountByClient(String email);
    double getTotalPriceByEmail(String email);
    void updateProductQuantity(String user_email, Long productId, int quantity);
    boolean isProductInCart(Cart cart);
}














